### PR TITLE
Edit restaurants & fix up show

### PR DESCRIPTION
**Pull request by ghost**
PT Story: https://www.pivotaltracker.com/n/projects/1878955

Changes proposed in this pull request:
- Created restaurant edit page
- Gave it a form and button
- Restaurant owner can now edit restaurant info

What I have learned working on this feature:
- How to make a form DRY, by making a partial
- How to refactor
- How to reload
- How to create an error message in different ways

Screenshots: 
skarmavbild 2016-10-04 kl 16 27 01

Ready for review!

### CODE REVIEW

**Comment by tochman** [comment]
Is everyone onboard with extracting the authorization part of this story?

**Comment by PetronellaSimonsbacka** [comment]
Yes they are. I talked to theSuss, Jesper and Amber for me and Victoria, so
everybody are omboard. I extracted it and made it a feature worth one point
and Amber changed it to a chore. It is all good and in the interest of
progress. Me and Victoria will continue on it.
/Nella

2016-10-04 17:06 GMT+02:00 Thomas Ochman notifications@github.com:

File: db/schema.rb [hunk 1]
```diff
@@ -15,6 +15,16 @@
...
 
+  create_table "add_menies", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  create_table "add_menus", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
```
Comment (tochman) [inline db/schema.rb]: What's this??

File: db/schema.rb [hunk 2]
```diff
@@ -39,6 +49,11 @@
     t.datetime "updated_at",              null: false
   end
 
+  create_table "orders", force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
```
Comment (tochman) [inline db/schema.rb:43]: What's this??

File: features/step_definitions/restaurant_steps.rb [hunk 3]
```diff
@@ -18,6 +18,32 @@
...
+Then(/^I should be on the show page for "([^"]*)"$/) do |restaurant|
+  restaurant_id = Restaurant.find_by(name: restaurant)
+  expect(current_path).to eq restaurant_path(id: restaurant_id)
+end
+
+
+#Then(/^I should be on the "([^"]*)" page for "([^"]*)"$/) do |page, restaurant|
+#  restaurant_id = Restaurant.find_by(name: restaurant)
+#  expect(current_path).to eq edit_restaurant_path(id: restaurant_id)
+#end
```
Comment (tochman) [inline features/step_definitions/restaurant_steps.rb]: Plz remove comments

**Review by tochman — CHANGES_REQUESTED** [review]

**Comment by PetronellaSimonsbacka** [comment]
Hi Thomas!
I have removed the comments, but do not know what the stuff in schema
is???????
The pull request was made from Victorias computer, she is doing her nails
now, so I will slack her and say that I cleaned up the file and then she
could either also take it away or pull from mine and upload.
/Nella

2016-10-04 17:10 GMT+02:00 Thomas Ochman notifications@github.com:

File: db/schema.rb (hunk 1 above)
Comment (ghost) [reply db/schema.rb]: I have no idea? We haven't touched that file? Why would we? We've been working on restaurants.. Not menus?

File: db/schema.rb (hunk 2 above)
Comment (ghost) [reply db/schema.rb:43]: Same goes for this one?

File: features/step_definitions/restaurant_steps.rb (hunk 3 above)
Comment (tochman) [reply features/step_definitions/restaurant_steps.rb]: Again, please remove comments before merge

File: features/restaurant_edit.feature [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  Scenario: I edit my restaurant
+    Given I am on the restaurant page for "Awesome"
+    And I click the link "Edit"
+    When I fill in "Name" with "tacos"
+    And I click the "Submit" button
+    Then I should be on the show page for "tacos"
+    And I should see "tacos"
+
+  Scenario: I fuck up the edit
```
Comment (tochman) [inline features/restaurant_edit.feature]: I know I wrote this but please change the scenario title 😉

File: features/step_definitions/restaurant_steps.rb [hunk 5]
```diff
@@ -18,6 +18,32 @@
...
+end
+
+Given(/^I am on the restaurant page for "([^"]*)"$/) do |name|
+  restaurant = Restaurant.find_by(name: name)
+  visit(restaurant_path(restaurant))
+end
+
+Then(/^I should be on the edit restaurant page for "([^"]*)"$/) do |restaurant|
+  restaurant_id = Restaurant.find_by(name: restaurant)
+  expect(current_path).to eq edit_restaurant_path(id: restaurant_id)
```
Comment (tochman) [inline features/step_definitions/restaurant_steps.rb]: Just restaurant is a better variable name

**Review by tochman — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Edit restaurants & fix up show: 
- "plz merge me"

File: features/step_definitions/restaurant_steps.rb (hunk 3 above)
Comment (tochman) [reply features/step_definitions/restaurant_steps.rb]: Comments

**Review by tochman — APPROVED** [review]

**Pull request merged**